consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -723,6 +1093,87 @@ impl PruningProofManager {
             .collect_vec()
     }
 
+    /// BFS forward iterates from genesis_hash until selected tip, ignoring blocks in the antipast of selected_tip.
+    /// For each block along the way, insert that hash into the ghostdag_store
+    /// If we have a required_block to find, this will return true if that block was found along the way
+    fn fill_level_proof_ghostdag_data(
+        &self,
+        genesis_hash: Hash,
+        selected_tip: Hash,
+        ghostdag_store: &Arc<DbGhostdagStore>,
+        required_block: Option<Hash>,
+        level: BlockLevel,
+    ) -> bool {
+        let relations_service = RelationsStoreInFutureOfRoot {
+            relations_store: self.level_relations_services[level as usize].clone(),
+            reachability_service: self.reachability_service.clone(),
+            root: genesis_hash,
+        };
+        let gd_manager = GhostdagManager::new(
+            genesis_hash,
+            self.ghostdag_k,
+            ghostdag_store.clone(),
+            relations_service.clone(),
+            self.headers_store.clone(),
+            self.reachability_service.clone(),
+            level != 0,
+        );
+
+        ghostdag_store.insert(genesis_hash, Arc::new(gd_manager.genesis_ghostdag_data())).unwrap();
+        ghostdag_store.insert(ORIGIN, gd_manager.origin_ghostdag_data()).unwrap();
+
+        let mut topological_heap: BinaryHeap<_> = Default::default();
+        let mut visited = BlockHashSet::new();
+        for child in relations_service.get_children(genesis_hash).unwrap().read().iter().copied() {
+            topological_heap.push(Reverse(SortableBlock {
+                hash: child,
+                // It's important to use here blue work and not score so we can iterate the heap in a way that respects the topology
+                blue_work: self.headers_store.get_header(child).unwrap().blue_work, // TODO: Maybe add to compact data?
```

Comment:
I think this TODO can be removed. Adding blue work to compact data would be a DB version change and there's no need to specify this TODO here